src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketJobLinkActionFactory.java

```diff
@@ -0,0 +1,72 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import hudson.Extension;
+import hudson.model.Action;
+import hudson.model.FreeStyleProject;
+import hudson.model.Job;
+import hudson.util.FormValidation;
+import jenkins.model.TransientActionFactory;
+import org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+
+@Extension
+public class BitbucketJobLinkActionFactory extends TransientActionFactory<Job> {
```

Comment:
The key change here is the action factory is now opened for all _jobs_, not just projects. I'm not 100% with Jenkins standard, if they prefer more specific or generic action factories, and happy to produce two action factories for each job if that's preferred.